input/postgres/server_stats.go

```diff
@@ -21,12 +21,64 @@ SELECT
 FROM pg_catalog.pg_control_checkpoint()
 `
 
+const xminHorizonSQL string = `
+WITH slots AS (
+	SELECT
+		xmin,
+		catalog_xmin
+	FROM pg_replication_slots
+	WHERE xmin IS NOT NULL
+	ORDER BY age(xmin) DESC
+	LIMIT 1
+)
+SELECT
+(
+	SELECT
+		backend_xmin
+	FROM pg_stat_activity
+	WHERE backend_xmin IS NOT NULL OR backend_xid IS NOT NULL
+	ORDER BY greatest(age(backend_xmin), age(backend_xid)) DESC
+	LIMIT 1
+) as backend,
+(
+	SELECT xmin FROM slots
+) as replication_slot_xmin,
+(
+	SELECT catalog_xmin FROM slots
+) as replication_slot_catalog_xmin,
+(
+	SELECT
+		transaction AS xmin
+	FROM pg_prepared_xacts
+	ORDER BY age(transaction) DESC
+	LIMIT 1
+) as prepare_xact,
+(
+	SELECT
+		backend_xmin
+	FROM pg_stat_replication
```

Comment:
I think we may need to handle our `get_stat_replication` helper function here, as described in our documentation (e.g. on https://pganalyze.com/docs/install/self_managed/01_create_monitoring_user) and already handled that way in `input/postgres/replication.go`.

That said, as a side note, I've been meaning for a while to review whether we actually need to ask people to create that helper. Re-reading things online, `pg_monitor` should include `pg_stat_replication`. So maybe this is outdated and we should actually drop that helper function from our docs? (it could still be helpful to support it here, for consistency with the other file)

Also, on the note of permissions, I wonder if we have any permission problems on e.g. systems like Heroku Postgres that don't give access to `pg_monitor`? (i.e. what happens when you query these tables there?)

Reply:
Noted on `get_stat_replication`, regardless of potentially dropping it in the future, I think it's good to be consistent to the current code, so I'll make a tweak to support it 👍

Re: permission, good call, I'll check with Heroku Postgres and see if this query goes well.

Reply:
I checked with Heroku Postgres and Aurora Postgres, there were no issue running this query with both of them. I also checked with a brand new user with pretty much zero permission but it can still run this query ✨ 

btw, for Heroku, with `\du`, I was able to see that the user is a member of `pg_monitor`. Is "systems like Heroku Postgres that don't give access to `pg_monitor`" still true?

Reply:
Thanks for checking! And maybe this is no longer true for Heroku. I recall Aiven might be another provider that does not give `pg_monitor` access. But I don't think we need to do further testing for now (AFAIK the way insufficient privileges would show here is that we just don't get any rows, which seems fine).